[readtapi] Use OptParser.td for options

cyndyishida opened the pull request:
This includes revamping how --compare was handled and adds `-o` ontop of tablegen approach.

This will be used to add more complex options.

ributzka commented on llvm/test/tools/llvm-readtapi/command-line.test:
```diff
@@ -0,0 +1,7 @@
+CHECK: OVERVIEW: LLVM TAPI file reader and manipulator
+CHECK: USAGE: llvm-readtapi [options] <inputs>
+CHECK: OPTIONS:
+CHECK:   -help    display this help
```
Does it also emit help for the compare option?

cyndyishida replied:
Yes, I didn't add the others so I wouldn't need to update it as more options get added.

ributzka commented on llvm/tools/llvm-readtapi/Opts.td:
`Opts.td` is a very generic file name. Could this be renamed to for example `TapiOpts.td`?

cyndyishida replied:
Yes, I was mostly following a pattern set by other tools. But it's not super consistent anyway. 

```
/Projects/llvm/llvm-project/llvm/tools
$ fd Opts.td
llvm-cvtres/Opts.td
llvm-cxxfilt/Opts.td
llvm-debuginfod/Opts.td
llvm-dwp/Opts.td
llvm-gsymutil/Opts.td
llvm-ifs/Opts.td
llvm-libtool-darwin/Opts.td
llvm-lipo/LipoOpts.td
llvm-ml/Opts.td
llvm-mt/Opts.td
llvm-nm/Opts.td
llvm-objcopy/BitcodeStripOpts.td
llvm-objcopy/CommonOpts.td
llvm-objcopy/InstallNameToolOpts.td
llvm-objcopy/ObjcopyOpts.td
llvm-objcopy/StripOpts.td
llvm-objdump/ObjdumpOpts.td
llvm-objdump/OtoolOpts.td
llvm-rc/Opts.td
llvm-rc/WindresOpts.td
llvm-readobj/Opts.td
llvm-readtapi/Opts.td
llvm-size/Opts.td
llvm-strings/Opts.td
llvm-symbolizer/Opts.td
llvm-tli-checker/Opts.td
sancov/Opts.td
```

ributzka commented on llvm/tools/llvm-readtapi/llvm-readtapi.cpp:
```diff
@@ -11,80 +11,132 @@
-      cl::PrintHelpMessage();
-      return EXIT_FAILURE;
-    }
+const int NON_TAPI_EXIT_CODE = 2;
```
What is `NON_TAPI_EXIT_CODE`?

cyndyishida replied:
`readtapi --compare` behaves like `diff` where exit codes differentiate between the same/different/other kind of error (e.g. missing file) 
I can add a comment to make that clearer.

ributzka reviewed (approved):
LGTM. Thanks Cyndy

cyndyishida committed:
[readtapi] Use OptParser.td for options

his includes revamping how --compare was handled and adds `-o` ontop of tablegen approach.

This will be used to add more complex options.
